### PR TITLE
Blocks: Simple explorer plugin for Blocks design

### PR DESCRIPTION
**Pull request by kanishkan**
Provides Blocks CGRA design support. Similar to VLIWConnectIC plugin, the BlocksConnectIC plugin is added to the GUI of ProDe and Explorer framework.
- The proposed plugin enforces connectivity requirements of Blocks architecture. i.e. It Modifies the IC network of given ADF to Blocks-Interconnect.
-  Resources (FU, RF, IMM, Memory) are not touched by the plugin
- Current support is limited to 32-bit Blocks IC design

### CODE REVIEW

File: tce/explorer/BlocksConnectIC.cc [hunk 1]
```diff
@@ -0,0 +1,266 @@
...
+    THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+    IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+    FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+    THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+    LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+    FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+    DEALINGS IN THE SOFTWARE.
+ */
+/**
+ * @file BlocksConnectIC.cc (Derrived from VLIWConnectIC.cc)
```
Comment (pjaaskel) [inline tce/explorer/BlocksConnectIC.cc:25]: typo derrived. also comma missing from the brief comment.

File: tce/explorer/BlocksConnectIC.cc [hunk 2]
```diff
@@ -0,0 +1,266 @@
...
+    requiresSimulationData() const {
+        return false;
+    }
+    virtual bool
+    requiresApplication() const {
+        return false;
+    }
+
+    /**
+     * Arranges architecture FUs into a Blocks-like interconnection. This is
```
Comment (pjaaskel) [inline tce/explorer/BlocksConnectIC.cc:79]: Brief comment, separate by empty line. Perhaps add reference URL for Blocks?

**Review by pjaaskel — APPROVED** [review]
just nitpicks, can ne fixed post merge

File: tce/explorer/BlocksConnectIC.cc (hunk 2 above)
Comment (kanishkan) [reply tce/explorer/BlocksConnectIC.cc]: Added Mark's thesis!

File: tce/explorer/BlocksConnectIC.cc (hunk 1 above)
Comment (kanishkan) [reply tce/explorer/BlocksConnectIC.cc]: Fixed typo!

**Comment by kanishkan** [comment]
@pjaaskel : Thanks for your reviews.
Fixed the comments, and this passes the system test.

**Review by pjaaskel — APPROVED** [review]

**Pull request merged**